### PR TITLE
feat: add news plugin

### PR DESCRIPTION
**Pull request by xiaohai-78**

### CODE REVIEW

File: community/plugins/spring-ai-alibaba-starter-plugin-news/pom.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	<artifactId>spring-ai-alibaba-starter-plugin-news</artifactId>
+	<name>spring-ai-alibaba-starter-plugin-news</name>
+	<description>News tool for Spring AI Alibaba</description>
+	<url>https://github.com/alibaba/spring-ai-alibaba</url>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework.ai</groupId>
+			<artifactId>spring-ai-spring-boot-autoconfigure</artifactId>
+		</dependency>
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-news/pom.xml:21]: please add space for each dependency

File: community/plugins/spring-ai-alibaba-starter-plugin-news/pom.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-webflux</artifactId>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-news/pom.xml:40]: please add fileName and remove other plugins irrelevant

File: community/plugins/spring-ai-alibaba-starter-plugin-news/src/main/java/com/alibaba/cloud/ai/plugin/news/NewsAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Description;
+
+/**
+ * @Author: XiaoYunTao
+ * @Date: 2024/12/18
+ */
+@Configuration
+@ConditionalOnClass({ SinaService.class, ToutiaoService.class })
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-news/src/main/java/com/alibaba/cloud/ai/plugin/news/NewsAutoConfiguration.java:17]: add ConditionalonProperty annotation in order to implement free loading

File: community/plugins/spring-ai-alibaba-starter-plugin-news/src/main/java/com/alibaba/cloud/ai/plugin/news/NewsAutoConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+public class NewsAutoConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean
+	@Description("Get the news from the toutiao news (获取今日头条新闻).")
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-news/src/main/java/com/alibaba/cloud/ai/plugin/news/NewsAutoConfiguration.java:22]: each bean should end with Function

File: community/plugins/spring-ai-alibaba-starter-plugin-news/src/main/java/com/alibaba/cloud/ai/plugin/news/NewsService.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * @Author: XiaoYunTao
+ * @Date: 2024/12/19
+ */
+public abstract class NewsService<T extends NewsService.Request, R extends NewsService.Response>
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-news/src/main/java/com/alibaba/cloud/ai/plugin/news/NewsService.java:17]: it should use XxxNewsService better, and does not use genericity

File: spring-ai-alibaba-examples/function-calling-example/pom.xml [hunk 6]
```diff
@@ -90,6 +90,11 @@
 			<version>${spring-ai-alibaba.version}</version>
 		</dependency>
 
+		<dependency>
+			<groupId>com.alibaba.cloud.ai</groupId>
+			<artifactId>spring-ai-alibaba-starter-plugin-news</artifactId>
```
Comment (PolarishT) [inline spring-ai-alibaba-examples/function-calling-example/pom.xml:95]: we changed plugin module into functioncalling module

```
<dependency>
<groupId>com.alibaba.cloud.ai</groupId>
<artifactId>spring-ai-alibaba-starter-function-calling-news</artifactId>
<dependency>
```

**Review by PolarishT — COMMENTED** [review]
for building more atom modules, we encourage contributors to separate news functionCalling module into two modules.
for example, **spring ai alibaba sina module** and **spring ai alibaba toutiao module**

**Comment by xiaohai-78** [comment]
done

File: pom.xml [hunk 7]
```diff
@@ -45,6 +45,8 @@
 		<module>community/plugins/spring-ai-alibaba-starter-plugin-gaode</module>
         <module>community/plugins/spring-ai-alibaba-starter-plugin-weather</module>
         <module>community/plugins/spring-ai-alibaba-starter-plugin-larksuite</module>
+        <module>community/plugins/spring-ai-alibaba-starter-function-calling-sinanews</module>
+        <module>community/plugins/spring-ai-alibaba-starter-function-calling-toutiaonews</module>
```
Comment (PolarishT) [inline pom.xml:49]: 这里怎么是 plugins ？应该是 funcationCalling 吧～ 是不是忘记更新代码啦～

**Review by PolarishT — CHANGES_REQUESTED** [review]

File: spring-ai-alibaba-examples/function-calling-example/pom.xml [hunk 8]
```diff
@@ -90,6 +90,17 @@
 			<version>${spring-ai-alibaba.version}</version>
 		</dependency>
 
+		<dependency>
```
Comment (yuluo-yx) [inline spring-ai-alibaba-examples/function-calling-example/pom.xml:93]: example module 已经移除了，这部分可以删掉，然后在 exmaple 仓库提交一下？

Comment (xiaohai-78) [reply spring-ai-alibaba-examples/function-calling-example/pom.xml:93]: ok

**Pull request closed without merging**